crates/knope-versioning/src/versioned_file/text_file.rs

```diff
@@ -0,0 +1,261 @@
+#[cfg(feature = "miette")]
+use miette::Diagnostic;
+use regex::Regex;
+use relative_path::RelativePathBuf;
+use thiserror::Error;
+
+use crate::{Action, semver::Version};
+
+#[derive(Clone, Debug)]
+pub struct TextFile {
+    pub(super) path: RelativePathBuf,
+    content: String,
+    pattern: String,
+    diff: Option<String>,
+}
+
+impl TextFile {
+    /// Creates a new TextFile with the given pattern.
+    ///
+    /// # Errors
+    ///
+    /// If the regex pattern is invalid
+    pub fn new(path: RelativePathBuf, content: String, pattern: String) -> Result<Self, Error> {
+        // Validate the regex pattern
+        Regex::new(&pattern).map_err(|source| Error::InvalidPattern {
```

Comment:
Probably better to store the compiled version instead of needing to recompile later